docs/reference/accounts.md

```diff
@@ -27,7 +29,11 @@ Constraints:
 
 * Type is 128-bit unsigned integer (16 bytes)
 * Must not be zero or `2^128 - 1` (the highest 128-bit unsigned integer)
-* Must not conflict with another account
+* Must not conflict with another account (note that an account MAY have the same `id` as a transfer,
+but we would recommend avoiding this)
```

Comment:
We should always explain why. i.e. Why we recommend avoiding ID collisions across namespaces. For example, other systems may not allow this, therefore being stricter in what one sends to TigerBeetle improves interoperability.

At this point, the question will arise, well... why does TigerBeetle allow it then? And the answer is that if one is migrating from SQL to TigerBeetle, then I think there may be Account and Transfer tables with monotonically increasing integer IDs that collide. Of course, the user could fix this up by hashing these to derive UUIDs that are unique across namespaces.

Reply:
I'm going to remove this note and just link to the `id` section on the data modeling page, because that already includes a mention of this